setup.py

```diff
@@ -42,7 +42,7 @@
         "google-cloud-bigquery-storage >= 0.6.0, <2.0.0dev",
         # Bad Linux release for 0.14.0.
         # https://issues.apache.org/jira/browse/ARROW-5868
-        "pyarrow>=0.13.0, != 0.14.0",
+        "pyarrow>=0.16.0, < 1.0dev",
```

Comment:
Do we really want to restrict ourselves to versions prior to the first stable version, i.e. 1.0.0?

Going from 0.x to 1.0 normally represents that the library achieved certain level of stability and maturity, unlike bumping an existing major version from, say, 1.x to 2.0.

Should perhaps the upper bound instead be set to `< 2.0dev`?

Reply:
Agreed. We should either restrict to `<0.17.0dev` or `<2.0.0dev`. Since the pyarrow team is planning to be only a release or two from 1.0, I suggest `<2.0.0dev`.

Reply:
Also, the comment about 0.14.0 can be removed.